### PR TITLE
refactor: naming changes - liftN families of functions are now mapN (issue #5474)

### PR DESCRIPTION
**Pull request by stephentetley**
This PR changes the naming of "liftN" families of functions to be "mapN".

The lift families are changed in:
Applicative
Boxable
List
Option 
Result
Validation

The Boxable changes mandate a change to `Lowering.scala` in the compiler. The changes to the lines with `mkDefnSym` are necessary, the other changes are cosmetic so the scala functions aren't called `liftX` and `liftXb` when they generate code that now uses `Boxable.map`. 

To be honest I'm not sure if the cosmetic changes are worthwhile - I'm happy to revert them...

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Fast service!

I think we should leave the `Boxable` functions alone because their signatures are different from the applicative lifts. In particular, they don't take any arguments like the `mapN` functions.

**Comment by stephentetley** [comment]
I'll revert all of `Boxable` this evening after work.

**Comment by stephentetley** [comment]
`Boxable` should now be reverted if the tests pass.

File: main/src/library/Applicative.flix [hunk 1]
```diff
@@ -50,28 +50,28 @@ pub lawful class Applicative[m : Type -> Type] with Functor[m] {
     /// Instances can define more efficient implementations than the default implementation
     /// (which is `Applicative.ap(Functor.map(f, x1), x2)`).
     ///
-    pub def liftA2(f: t1 -> t2 -> r \ ef, x1: m[t1], x2: m[t2]): m[r] \ ef = Applicative.ap(Functor.map(f, x1), x2)
+    pub def mapA2(f: t1 -> t2 -> r \ ef, x1: m[t1], x2: m[t2]): m[r] \ ef = Applicative.ap(Functor.map(f, x1), x2)
```
Comment (magnus-madsen) [inline main/src/library/Applicative.flix:53]: I think these should be `map2` without the `A` (at least to follow the Scala cats naming we are using).

Comment (stephentetley) [reply main/src/library/Applicative.flix:53]: Right - will fix.

Comment (stephentetley) [reply main/src/library/Applicative.flix:53]: There's also a `lift` function in `Functor` that is just `Functor.map`.

Do we want to remove it - or look at it another time?

Finally `Functor.as` shares its name with a keyword - would it be better called `Functor.replace`?

Comment (magnus-madsen) [reply main/src/library/Applicative.flix]: Yes to both i think

Comment (stephentetley) [reply main/src/library/Applicative.flix]: Both are done in the latest commit.

**Comment by magnus-madsen** [comment]
Fixes #5474

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**